Automattic/wp-calypso
# Post Editor: Extract "View Link" component from Editor Action Bar

For a published post, there is a "View Link" button in the editor action bar than opens the post in a new tab:
<img width="182" alt="screen shot 2018-05-21 at 15 20 25" src="https://user-images.githubusercontent.com/664258/40310220-599d4016-5d0c-11e8-9880-210f4e1b2800.png">

This PR seeks to extract the code from `EditorActionBar` into a separate component that fetches the post and post type date from Redux.

I also merge the `ViewLabel` component, that merely builds the "View Page"/"View Post"/"View CPT" label for the tooltip, into the main `ViewLink` component.

There's no longer a need to pass the Flux-based `savedPost` prop to `EditorActionBar`. That was the main goal of this refactoring.

**How to test:**
Check that the "View Link" button is shown when editing a published post and that it opens the correct URL. Verify that the label refers to the right post type. For example, in the screenshot above, I'm editing a "Testimonial" CPT.
